Framework/DataHandling/src/LoadPSIMuonBin.cpp

```diff
@@ -189,14 +193,47 @@ void LoadPSIMuonBin::exec() {
 
   // If timeZero is bigger than the largest bin assume it refers to a bin's
   // value
+  double absTimeZero = timeZero;
   if (timeZero > largestBinValue) {
-    setProperty("TimeZero",
-                outputWorkspace->x(0)[static_cast<int>(std::floor(timeZero))]);
-  } else {
-    setProperty("TimeZero", timeZero);
+    absTimeZero = outputWorkspace->x(0)[static_cast<int>(std::floor(timeZero))];
+  }
+  setProperty("TimeZero", absTimeZero);
+
+  auto firstGoodDataSpecIndex = static_cast<int>(
+      *std::max_element(m_header.firstGood, m_header.firstGood + 16));
+
+  setProperty("FirstGoodData", outputWorkspace->x(0)[firstGoodDataSpecIndex]);
+
+  // Time zero is when the pulse starts.
+  // The pulse should be at t=0 to be like ISIS data
+  // manually offset the data
+  for (auto specNum = 0u; specNum < m_histograms.size(); ++specNum) {
+    auto xData = outputWorkspace->mutableX(specNum);
```

Comment:
Your taking a copy here hence we need to reassign below, as the API (MatrixWorkspace.h:263) supports be ref we should do
`auto &xData = ...`